rs/tests/system_tests.bzl

```diff
@@ -33,26 +33,33 @@ def _run_system_test(ctx):
         content = """#!/bin/bash
             set -eEuo pipefail
 
-            # for every ic-os image specified, first ensure it's in remote storage, then
-            # export its download URL as an environment variable.
-            if [ -n "$ICOS_IMAGES" ]; then
-              # split the ";"-delimited list of "envvar:filepath;envvar2:filepath2;..."
+            # Resolve any RUN_SCRIPT_ variables
+
+            # RUN_SCRIPT_REPO_ICOS_IMAGES:
+            # For every repo image specified, first ensure it's in remote
+            # storage, then export its download URL and HASH as environment
+            # variables.
+            if [ -n "$RUN_SCRIPT_REPO_ICOS_IMAGES" ]; then
+              # split the ";"-delimited list of "env_prefixr:filepath;env_prefix2:filepath2;..."
               # into an array
-              IFS=';' read -ra icos_images <<<"$ICOS_IMAGES"
-              for image in "${{icos_images[@]}}"; do
+              IFS=';' read -ra repo_icos_images <<<"$RUN_SCRIPT_REPO_ICOS_IMAGES"
+              for image in "${{repo_icos_images[@]}}"; do
                   # split "envvar:filepath"
-                  image_varname=${{image%:*}}
+                  image_var_prefix=${{image%:*}}
                   image_filename=${{image#*:}}
 
                   # ensure the dep is uploaded
-                  image_download_url=$("$UPLOAD_SYSTEST_DEP" "$image_filename")
-                  echo "  -> $image_varname=$image_download_url" >&2
-
-                  # set the environment variable for the test
-                  export "$image_varname=$image_download_url"
+                  image_download_url=$("$RUN_SCRIPT_UPLOAD_SYSTEST_DEP" "$image_filename")
+                  echo "  -> $image_filename=$image_download_url" >&2
+
+                  # Since this is a CAS url, we assume the last URL path part is the sha256
+                  image_download_hash="${{image_download_url##*/}}"
+                  # set the environment variables for the test
+                  export "${{image_var_prefix}}_URL=$image_download_url"
+                  export "${{image_var_prefix}}_HASH=$image_download_hash"
```

Comment:
do we even need the hash? we could skip this entirely, assuming we trust SSL (and our object store)

Reply:
The hash is used from the test side of things to set up the proposals for upgrades. We could calculate the hash ourselves, but we don't always have the image locally, like when we want to test with the published mainnet url.